app/services/character/relocater.rb

```diff
@@ -0,0 +1,97 @@
+# Steps:
+# validate single user, unshared icons, galleries, and templates (if included), lack of character groups
+# collect: templates (if included), characters, posts, replies, drafts, galleries, icons
+# update all on unaudited (templates, aliases, drafts, galleries, icons)
+# UpdateModelJob on audited (characters, posts, replies)
+
+class Character::Relocater < Object
+  def initialize(admin_id)
+    @audited_user_id = admin_id
+  end
+
+  def transfer(character_ids, new_user_id, include_templates: false)
+    @characters = Character.where(id: character_ids)
+    @user_id = validate_user!
+    @user_other_characters = Character.where(user_id: @user_id).where.not(id: @characters.ids)
+    @other_character_galleries = get_galleries(@user_other_characters)
+    @other_character_icon_ids = get_icon_ids(@user_other_characters, @other_character_galleries)
+
+    raise CharacterGroupError.new('Characters must not have groups') if @characters.where.not(character_group: nil).exists?
+    @galleries = validate_galleries!
+    @icons = validate_icons!
+    @templates = validate_templates! if include_templates
+
+    @posts = Post.where(character_id: character_ids, user_id: @user_id)
+    @replies = Reply.where(character_id: character_ids, user_id: @user_id)
+    @drafts = ReplyDraft.where(character_id: character_ids, user_id: @user_id)
+
+    User.find(new_user_id)
+
+    # rubocop:disable Rails/SkipsModelValidations
+    @templates.update_all(user_id: new_user_id) if include_templates
+    @galleries.update_all(user_id: new_user_id)
+    @icons.update_all(user_id: new_user_id)
+    @drafts.update_all(user_id: new_user_id)
+    # rubocop:enable Rails/SkipsModelValidations
+
+    UpdateModelJob.perform_later(Post.to_s, { user_id: @user_id }, { user_id: new_user_id }, @audited_user_id)
+    UpdateModelJob.perform_later(Reply.to_s, { user_id: @user_id }, { user_id: new_user_id }, @audited_user_id)
+    UpdateModelJob.perform_later(Character.to_s, { user_id: @user_id }, { user_id: new_user_id }, @audited_user_id)
```

Comment:
Should we handle how this could impact PostAuthor too? e.g. if a character is transferred in a thread that didn't previously have this new author